src/rtctools/optimization/goal_programming_mixin_base.py

```diff
@@ -394,7 +394,17 @@ def __init__(self, optimization_problem):
         # Extract state range from model
         if self.has_target_bounds:
             try:
-                self.function_range = optimization_problem.bounds()[self.state]
+                # Handle ensemble-specific bounds by using ensemble_member=0 if
+                # feature flag is enabled
+                # TODO: What do we do with this? Max? Min? User explicit input?
```

Comment:
Why don't we check here if the bounds for this variable are the same for all members. If they are, we proceed as usual. If they are not, we raise an exception and instruct the caller to set `function_range` manually.

Reply:
Will still be a somewhat ugly check (support np.ndarray, Timeseries, and floats), but could work yes.

Reply:
Made something anyway; the user will just have to explicitly set something and then override it later I guess (if they're feeling fancy)